tests/test_common/test_system.py

```diff
@@ -74,180 +60,164 @@ def test_resolve_alias():
     assert str(mname.file) == "whatever/topvalysar--depth_surface.gri"
 
 
-@tsetup.skipifmac
-@tsetup.skipifwindows
-def test_xtgeocfile():
-    """Test basic system file io etc functions."""
-    gfile = xtgeo._XTGeoFile(TESTFILE)
-    xfile = xtgeo._XTGeoFile(TESTNOEXISTFILE)
-    yfile = xtgeo._XTGeoFile(TESTNOEXISTFOLDER)
-    gfolder = xtgeo._XTGeoFile(TESTFOLDER)
+@pytest.fixture
+def reek_grid_path(testpath):
+    return pathlib.Path(testpath) / "3dgrids/reek"
+
+
+@pytest.mark.parametrize(
+    "filename, does_exist",
+    [
+        ("REEK.EGRID", True),
+        ("", True),
+        ("NOSUCH.EGRID", False),
+        ("NOSUCH/NOSUCH.EGRID", False),
+    ],
+)
+def test_file_exist(reek_grid_path, filename, does_exist):
+    xtgeo_file = xtgeo._XTGeoFile(reek_grid_path / filename)
+    assert xtgeo_file.exists() is does_exist
+
+
+@pytest.mark.parametrize(
+    "filename, check_ok",
+    [
+        ("REEK.EGRID", True),
+        ("NOSUCH.EGRID", False),
+        ("NOSUCH/NOSUCH.EGRID", False),
+    ],
+)
+def test_check_file(reek_grid_path, filename, check_ok):
+    xtgeo_file = xtgeo._XTGeoFile(reek_grid_path / filename)
+    assert xtgeo_file.check_file() is check_ok
+
+    if not check_ok:
```

Comment:
Personally I am not a fan of this pattern, as this is a very hard fork in the test. Would it not be better to split this into two tests? One `test_valid` and one `test_invalid`? Then you can check that the exit message is what you expect and get a more explicit test.

Reply:
I think I understood what you wanted and it smacks of clean codes "avoid boolean flags" so I agree :)